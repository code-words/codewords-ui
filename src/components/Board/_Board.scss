.Board {
  display: flex;
  flex-direction: column;
  align-items: center;

  .turn-status {
    margin-bottom: 12px;
    margin-top: -20px;
  }
}

.gameboard {
  background: rgb(21, 21, 24);
  border-radius: 8px;
  display: grid;
  padding: 5px;
  grid-template-columns: repeat(5, 1fr);
  grid-auto-rows: repeat(5, 1fr);
  grid-gap: 5px;
  border: 3px solid rgb(62, 85, 104);
}

.case-file {
  display: flex;
  align-items: center;
  height: 90px;
  width: 130px;
  border-radius: 5px;
  background: $intel-base;
}

.case-title {
  background: #FFFFFF;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  .word-txt {
    text-align: center;
    text-transform: uppercase;
    font-size: .9em;
    font-weight: 500;
    padding: 10px 0;
    color: rgb(108, 93, 74);
  }
}

.Active {
  color: $player-active;
}