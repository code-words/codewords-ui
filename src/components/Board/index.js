import React, {Component} from 'react';
// 
import {words} from '../../mock/mockData';
// 

const Board = ({ pType }) => {
  return (
    <section className="gameboard">
      {words.map(w => {
        return (
          <div className="case-file">
            <p className="case-title">
              {w}
            </p>
          </div>
        )
      })}
    </section>
  )
}

export default Board;