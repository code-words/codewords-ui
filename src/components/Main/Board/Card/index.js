import React, { Component } from 'react';
class Card extends Component {
  handleClick = e => {
    if (this.props.isActive && !this.props.card.type) {
      this.props.sendGuess({ id: e.currentTarget.id });
    }
  };
  
	render() {
		const { card, isActive, isHover } = this.props;
		console.log(isHover)
    const { type, flipped, word, id } = card;
		const intelCls = type ? type : '';
		const clickCls = !type && isActive && !flipped ? 'clickable' : null;
		const img = type ? require(`../../../../images/${type}.jpg`) : null;

		const content = flipped ? (
      <article className="card" key={`${word}`} id={id}>
				<img className="card" src={img} alt={type} />
			</article>
		) : (
			<article
				className={`card front ${intelCls} ${clickCls} ${isHover}-hover`}
				key={`${word}`}
				id={id}
				name={word}
				onClick={this.handleClick}>
				<div className="word-bg" key={`${word}-bg`}>
					<p className="word-txt" key={`${word}-txt`}>
						{word}
					</p>
				</div>
			</article>
		);

		return content;
	}
}

export default Card;
