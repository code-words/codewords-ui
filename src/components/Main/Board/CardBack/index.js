import React from 'react';

const CardBack = ({ word, type }) => {
  const url = `../../../../images/${type}.jpg`;

  return (
    <article
      className="card"
      key={`${word}-file`}
      style={{ backgroundImage: `url(${url})` }}
    >
    </article>
  );
};

export default CardBack;