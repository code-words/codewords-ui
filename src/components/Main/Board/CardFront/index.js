import React from 'react';

const CardFront = ({ card, isActive }) => {
  const intelClass = card.type ? card.type : '';
  const clickClass = !card.type && isActive ? 'clickable' : null;

  return (
    <article className={`card CardFront ${intelClass} ${clickClass}`} key={`${card.word}`}>
      <div className="word-bg" key={`${card.word}-bg`}>
        <p className="word-txt" key={`${card.word}-txt`}>{card.word}</p>
      </div>
    </article>
  )
};

export default CardFront;