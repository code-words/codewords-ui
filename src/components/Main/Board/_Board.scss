.Board {
  display: flex;
  flex-direction: column;
  align-items: center;

  .turn-status {
    margin-bottom: 12px;
    margin-top: -15px;
    font-size: 20px;
    font-weight: 300;
    font-family: $header-font;
  }
}

.gameboard {
  background: rgb(21, 21, 24);
  border-radius: 8px;
  display: grid;
  padding: 5px;
  grid-template-columns: repeat(5, 1fr);
  grid-auto-rows: repeat(5, 1fr);
  grid-gap: 5px;
  border: 3px solid rgb(62, 85, 104);
}

.card {
  display: flex;
  align-items: center;
  height: 84px;
  width: 130px;
  border-radius: 5px;
  background: $intel-base;
}

.Active {
  color: $player-active;
}

.Inactive {
  color: $player-inactive;
}