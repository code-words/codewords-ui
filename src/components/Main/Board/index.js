import React from 'react';
import {words} from '../../../mock/mockData';

const Board = props => {
  let status = "Active";

  console.log('Board Props', props)
  
  return (
    <section className="Board">
      <h2 className="turn-status">
        Agent Status:
        <span className={status}> {status}</span>
      </h2>
      <div className="gameboard">
        {props.data && props.data.map(card => {
          const intelClass = card.type ? card.type : '';

          return (
            <article className={`case-file ${intelClass}`} key={`${card}-file`}>
              <div className="case-title" key={`${card}-title`}>
                <p className="word-txt" key={`${card}-txt`}>{card}</p>
              </div>
            </article>
          )
        })}
      </div>
    </section>
  )
}

export default Board;