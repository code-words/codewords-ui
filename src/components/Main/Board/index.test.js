import React from 'react';
import Board from '.';
import { shallow } from 'enzyme';

describe('Board', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow(<Board />);
    instance = wrapper.instance();

  })

  it('should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})