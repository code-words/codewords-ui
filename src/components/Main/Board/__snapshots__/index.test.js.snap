// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Should render Component 1`] = `
"<Board userName=\\"test\\" cardData={{...}}>
  <section className=\\"Board\\">
    <h2 className=\\"turn-status\\">
      Agent (
      TEST
      ) Status:
      <span className=\\"Inactive\\">
         
        Inactive
      </span>
    </h2>
    <div className=\\"gameboard\\">
      <Card card={{...}} id={1} isHover={[undefined]} isActive={[undefined]} sendGuess={[undefined]}>
        <article className=\\"card front  null undefined-hover\\" id={1} name=\\"fml\\" onClick={[Function]}>
          <div className=\\"word-bg\\">
            <p className=\\"word-txt\\">
              fml
            </p>
          </div>
        </article>
      </Card>
    </div>
  </section>
</Board>"
`;

exports[`Board should match component snapshot 1`] = `ShallowWrapper {}`;
